app/forms/system_admin/user_lead_schools_form.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module SystemAdmin
+  class UserLeadSchoolsForm
+    include ActiveModel::Model
+
+    FIELDS = %i[
+      query
+      lead_school_id
+      search_results_found
+      results_search_again_query
+      no_results_search_again_query
+    ].freeze
+
+    QUERY_ERROR = I18n.t("activemodel.errors.models.user_lead_schools_form.attributes.query.length")
+
+    validates :lead_school_id, presence: true, if: :school_validation_required?
+
+    validates :query,
+              presence: true,
+              length: {
+                minimum: SchoolSearch::MIN_QUERY_LENGTH,
+                message: QUERY_ERROR,
+              },
+              if: -> { initial_search? }
+
+    validates :results_search_again_query,
+              presence: true,
+              length: {
+                minimum: SchoolSearch::MIN_QUERY_LENGTH,
+                message: QUERY_ERROR,
+              },
+              if: -> { results_searching_again? }
+
+    validates :no_results_search_again_query,
+              presence: true,
+              length: {
+                minimum: SchoolSearch::MIN_QUERY_LENGTH,
+                message: QUERY_ERROR,
+              },
+              if: -> { no_results_searching_again? }
+
+    attr_accessor(*FIELDS, :user)
+
+    def save!
```

Comment:
`save!` usually is used when the method raises error, instead of returning false, i'd rename this to `save`